stdlib/Distributed/src/managers.jl

```diff
@@ -184,29 +195,61 @@ function launch_on_machine(manager::SSHManager, machine, cnt, params, launched,
     tval = get(ENV, "JULIA_WORKER_TIMEOUT", "")
 
     # Julia process with passed in command line flag arguments
-    cmds = """
-        cd -- $(shell_escape_posixly(dir))
-        $(isempty(tval) ? "" : "export JULIA_WORKER_TIMEOUT=$(shell_escape_posixly(tval))")
-        $(shell_escape_posixly(exename)) $(shell_escape_posixly(exeflags))"""
+    if shell == :posix
+        # ssh connects to a POSIX shell
+
+        cmdline_cookie = false
+        exeflags = `$exeflags --worker`
+        cmds = """
+            cd -- $(shell_escape_posixly(dir))
+            $(isempty(tval) ? "" : "export JULIA_WORKER_TIMEOUT=$(shell_escape_posixly(tval))")
+            $(shell_escape_posixly(exename)) $(shell_escape_posixly(exeflags))"""
+
+        # shell login (-l) with string command (-c) to launch julia process
+        remotecmd = shell_escape_posixly(`sh -l -c $cmds`)
+
+    elseif shell == :wincmd
+        # ssh connects to Windows cmd.exe
+
+        # Passing the cookie via ssh stdin currently hangs with
+        # Microsoft's Windows port of OpenSSH, so we do it via
+        # --worker on the command line for now, although that may
+        # cause the cookie to become visible to other users (e.g.
+        # using Process Explorer or WMIC).
+        cmdline_cookie = true
+        exeflags = `$exeflags --worker=$(cluster_cookie())`
+
+        any(c -> c == '"', exename) && throw(ArgumentError("invalid exename"))
+
+        remotecmd = shell_escape_windows(exename, exeflags...)
+        if dir !== nothing && dir != ""
+            any(c -> c == '"', dir) && throw(ArgumentError("invalid dir"))
+            remotecmd = "pushd \"$(dir)\" && $remotecmd"
```

Comment:
I can't use `cd` because `cmd.exe` does not support setting the current working directory to a UNC path `\\filer\share\...`. I tried at first to get around this by changing the working directory via `julia -e "cd(\"\\\\filer\\share\...\")` only to realise that julia does not execute both options `-e` and `--worker` at the same time. I finally discovered that `pushd` is a much better command than `cd`, because when you give `pushd` a UNC, it automatically maps the share's root to a network drive as needed, and then `cmd.exe` happily takes that as its current working directory. (That discovery also reduced somewhat the need for very comprehensive string escaping that using option `-e` would have required, but having `cmd.exe` escaping implemented properly might still become useful one day if we put more stuff into `exeflags`.)

Reply:
Since Windows does not permit `"` in filenames, there is no need to escape any of them in the argument of `pushd`. Therefore, always quoting the pathname passed to `pushd` seemed sufficient here.

Reply:
"dosly" would be inaccurate because `cmd.exe` came new with Windows NT and is different from MS-DOS's `COMMAND.EXE`. (I don't recall if MS-DOS also had `^` escaping in `COMMAND.EXE`, it has been ~18 years since the last MS-DOS release ...)

Reply:
Ah, true. Wikipedia mentions that capability as an improvement of cmd.exe over COMMAND.COM. Even so, there may or might not be another special rule (for '%'): https://en.wikibooks.org/wiki/Windows_Batch_Scripting#Quoting_and_escaping